src/text-editor.coffee

```diff
@@ -609,15 +609,19 @@ class TextEditor extends Model
   # * "<filename>" when it is the only editing buffer with this file name.
   # * "<filename> — <unique-dir-prefix>" when other buffers have this file name.
   #
+  # * `textEditors` An optional {Array} of the other {TextEditor} instances that
```

Comment:
I wonder if we can take a slightly different design direction for this. Namely, I am concerned about keeping the `get...Title` responsibility within `TextEditor`: these methods seem to belong to UI models rather than to this object, which ultimately should be just a wrapper around `TextBuffer` + `TokenizedBuffer`. Therefore, how would you feel about extracting a `TextEditorPaneItem` object? It could look similar to this:

```js
class TextEditorPaneItem {
  getTitle ()
  getLongTitle ()
  getPath ()
  // all the other methods that belong to the `PaneItem` interface (which I couldn't find in the docs)
}
```

We could make this change backwards compatible in two steps:

1. Whenever a user tries to call `TextEditor`'s methods on `TextEditorPaneItem`'s prototype, forward them to the underlying editor instance and emit a deprecation.
2. Whenever a user tries to call `TextEditorPaneItem` methods on `TextEditor`'s prototype, forward them to `atom.workspace.paneForItem(this)` and emit a deprecation.

I feel like this solution could be what we might adopt long-term, but I understand this design decision has a larger impact on the users of this APIs, especially because calling `atom.workspace.open` returns a `TextEditor` instance, which means people would get quite a lot of deprecations if it starts returning `TextEditorPaneItem` instances. But I also feel like we should take this as an opportunity to improve Atom's internals design, without being afraid of changing them too much.

What do you think? 💭